File: frontend/types/product.ts
export interface Product {
  id: number;
  name: string;
  description: string;
  price: number;
  formattedPrice: string;
  stock: number;
  imageUrl: string;
  imageUrl2?: string;
  imageUrl3?: string;
  imageUrl4?: string;
  specifications?: string; // JSON string
  categoryId: number;
  categoryName: string;
}

export interface CreateProductDto {
  name: string;
  description: string;
  price: number;
  stock: number;
  imageFiles?: File[];
  specifications?: string; // JSON string
  categoryId: number;
}

File: Backend/Models/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Backend.Models
{
    public class Product
    {
        public int Id { get; set; }
        [Required]
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        [Column(TypeName = "decimal(18,2)")]
        public decimal Price { get; set; }
        public int Stock { get; set; }
        public string ImageUrl { get; set; } = string.Empty;
        public string? ImageUrl2 { get; set; }
        public string? ImageUrl3 { get; set; }
        public string? ImageUrl4 { get; set; }
        
        // Stores dynamic key-value pairs as JSON string
        public string? Specifications { get; set; }

        public int CategoryId { get; set; }
        public Category Category { get; set; } = null!;
    }
}
